tests/test_runner.py

```diff
@@ -39,3 +43,58 @@ def test(entrypoint, kwargs_list, expected_output):
     output = _runner(entrypoint, kwargs_list)
     # THEN
     assert serialize(output) == expected_output
+
+
+@pytest.mark.parametrize(
+    "entrypoint,file_contents,expected_output",
+    [
+        (
+            "tests.artifact_annotations.script_annotations_artifact_path:read_artifact",
+            "Hello there!",
+            "Hello there!",
+        ),
+        (
+            "tests.artifact_annotations.script_annotations_artifact_object:read_artifact",
+            """{"a": "Hello ", "b": "there!"}""",
+            "Hello there!",
+        ),
+        (
+            "tests.artifact_annotations.script_annotations_artifact_file:read_artifact",
+            "Hello there!",
+            "Hello there!",
+        ),
+    ],
+)
+def test_artifact(entrypoint, file_contents, expected_output, tmp_path, monkeypatch):
+    # GIVEN
+    if not tmp_path.is_file():
+        tmp_path = tmp_path / "my_file.txt"
+
+    tmp_path.touch()
+    tmp_path.write_text(file_contents)
+    import tests.helper as test_module
+
+    monkeypatch.setattr(test_module, "ARTIFACT_PATH", str(tmp_path))
+    kwargs_list = []
+    global_config.experimental_features["script_annotations"] = True
+    # WHEN
+    output = _runner(entrypoint, kwargs_list)
+    # THEN
+    assert serialize(output) == expected_output
+
+
+@pytest.mark.parametrize(
+    "entrypoint",
+    ["tests.artifact_annotations.script_annotations_artifact_wrong_loader:read_artifact"],
+)
+def test_artifact_wrong_loader(entrypoint):
+    # GIVEN
+    kwargs_list = []
+    global_config.experimental_features["script_annotations"] = True
```

Comment:
actual tests should be using the `global_config_fixture` now https://github.com/argoproj-labs/hera/blob/4bf33d05343efcc580fcb270a7e7ec951e51c556/tests/conftest.py#L7